common/serial_reader.h

```diff
@@ -0,0 +1,104 @@
+// USB-reader
+#pragma once
+
+#include <string>
+#include "termios.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <fstream>
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <chrono>
+#include <thread>
+#include <errno.h>
+
+#include <libserialport.h> // cross platform serial port lib
```

Comment:
Wait, are you using libserialport in this class anywhere? Or were you saying that it wasn't working for you?